### ISSUE TITLE
Finish the FermionicOperator implementation

### ISSUE THREAD

**Issue by manoelmarques** [Qiskit/qiskit-nature#21]
Migrated from Enterprise Github.
Creator: @mrossinek

- [x] implement the `to_opflow()` method in both, `primitives.FermionicOperator` and `FermionicSumOp`
- [x] unittest all methods

**Comment by manoelmarques** [comment]
Creator: @mrossinek

You can find an example of the new FermionicOperator in action here:
https://gist.github.ibm.com/oss-zurich/a2b75f58d4cd2eae7df8e1f1e9dcd6b0

**Comment by manoelmarques** [comment]
Creator: @mrossinek

For reference purposes I will add Simon's original code for the `to_qiskit()` method of the `FermionicOperator`. This can be used as a reference in comparison to the new `to_opflow()` method to guide the similar transition in the case of the other operator classes (#2, #3).
```python
    def to_qiskit(self, mode=default_mode, custom_transformed_XY = None) -> WeightedPauliOperator:
        """
        Returns the  transformation of `self` as a qiskit.aqua.operators WeightedPauliOperator object.
        """

        # Catch the exception of a zero operator:
        if len(self) == 0 or self.coeff == 0:
            return WeightedPauliOperator(paulis = [])

        # 0. Set up the lookup table `self.transformed_XY` for the JW transform
        if custom_transformed_XY is not None:
            self.transformed_XY = custom_transformed_XY

        if self.transformed_XY is None:
            if mode == 'jordan_wigner':
                self._jordan_wigner_mode()
            elif mode == 'bravyi_kitaev':
                self._bravyi_kitaev_mode()
            elif mode == 'parity':
                self._parity_mode()
            else:
                raise UserWarning("mode must be one of ['jordan_wigner', 'bravyi_kitaev', 'parity']")

        # 1. Initialize an operator list with the identity scaled by the `self.coeff`
        all_false = np.asarray([False] * len(self), dtype=np.bool)

        scaled_identity = WeightedPauliOperator(paulis=[[self.coeff,Pauli(z=all_false, x=all_false)]]) # self.coeff * I
        operator_list = [scaled_identity]

        # Go through the label and replace the fermion operators by their qubit-equivalent, then save the
        # respective Pauli string in the pauli_str list.
        for position, char in enumerate(self.label):
            # The creation operator is given by 0.5*(X + 1j*Y)
            if char == '+':
                real_part = [0.5, self.transformed_XY[position][0]]    # 0.5 * X
                imag_part = [0.5j, self.transformed_XY[position][1]]   # 0.5j * Y
                pauli_str = [real_part, imag_part]
                operator_list.append(WeightedPauliOperator(paulis=pauli_str))

            # The annihilation operator is given by 0.5*(X - 1j*Y)
            elif char == '-':
                real_part = [0.5, self.transformed_XY[position][0]]    # 0.5 * X
                imag_part = [-0.5j, self.transformed_XY[position][1]]  # -0.5j * Y
                pauli_str = [real_part, imag_part]
                operator_list.append(WeightedPauliOperator(paulis=pauli_str))

            # The occupation number operator N is given by 0.5*(I + Z)
            elif char == 'N':
                offset_part = [0.5, Pauli(z=all_false, x=all_false)]    # 0.5 * I
                z_part      = [0.5, self.transformed_XY[position][1] * self.transformed_XY[position][0]]  # 0.5 * YX
                pauli_str = [offset_part, z_part]
                operator_list.append(WeightedPauliOperator(paulis=pauli_str))

            # The `emptiness number` operator I - N is given by 0.5*(I - Z)
            elif char == 'E':
                offset_part = [0.5, Pauli(z=all_false, x=all_false)]     # 0.5 * I
                z_part      = [-0.5, self.transformed_XY[position][1] * self.transformed_XY[position][0]]  # -0.5 * Z
                pauli_str = [offset_part, z_part]
                operator_list.append(WeightedPauliOperator(paulis=pauli_str))

            elif char == 'I':
                continue

            # catch any disallowed labels
            else:
                raise UserWarning(
                    "BaseFermionOperator label included '{}'. Allowed characters: I, N, E, +, -".format(char))

        # 2. Multiply all transformed qubit operators to get the final result
        aqua_operator = operator_product(operator_list)
        # Remove pauli strings with 0 coefficient
        aqua_operator.chop()
        return aqua_operator
```

EDIT: Note, that the `N` and `E` operators were not working properly which is why they are now implemented slightly differently.

**Comment by mrossinek** [comment]
Closed by #25
